### ISSUE TITLE
[Bug]: Linux desktop client segfaults on Ubuntu 22.10

### ISSUE THREAD

**Issue by pgdooley** [nextcloud/desktop#4924]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

Latest update to the nextcloud desktop client crashes on loading with Segmentation fault (core dumped) on Ubuntu
22.10 with kernel 5.19.0-15-generic

### Steps to reproduce

1. Reboot system - client fails to load as normal
2. Invoke nextcloud client from command line produces segfault

### Expected behavior

Client loads and icon displayed in top bar

### Which files are affected by this bug

/usr/bin/nextcloud

### Operating system

Linux

### Which version of the operating system you are running.

Ubuntu 22.10 (development)

### Package

Distro package manager

### Nextcloud Server version

24.0.5

### Nextcloud Desktop Client version

3.6.0-1

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 3.3.6 to 3.4.0)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by akarl10** [comment]
this is the stack trace:
```
#0  doActivate<false>(QObject*, int, void**) (sender=0x0, signal_index=9, argv=0x7fffffffbc80) at kernel/qobject.cpp:3768
#1  0x00007fffeda9b607 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**)
    (sender=<optimized out>, m=m@entry=0x7fffee7f30c0 <QGuiApplication::staticMetaObject>, local_signal_index=local_signal_index@entry=1, argv=argv@entry=0x7fffffffbc80) at kernel/qobject.cpp:3946
#2  0x00007fffee266136 in QGuiApplication::screenAdded(QScreen*) (this=<optimized out>, _t1=<optimized out>, _t1@entry=0x555555af9830) at .moc/moc_qguiapplication.cpp:389
#3  0x00007fffee24a652 in QWindowSystemInterface::handleScreenAdded(QPlatformScreen*, bool) (ps=ps@entry=0x555555a77180, isPrimary=<optimized out>) at kernel/qwindowsysteminterface.cpp:826
#4  0x00007fffe8637c30 in QXcbConnection::initializeScreens() (this=this@entry=0x555555a66680) at ./src/plugins/platforms/xcb/qxcbconnection_screens.cpp:413
#5  0x00007fffe860f514 in QXcbConnection::QXcbConnection(QXcbNativeInterface*, bool, unsigned int, char const*)
    (this=0x555555a66680, nativeInterface=<optimized out>, canGrabServer=<optimized out>, defaultVisualId=4294967295, displayName=<optimized out>) at ./src/plugins/platforms/xcb/qxcbconnection.cpp:103
#6  0x00007fffe8612c69 in QXcbIntegration::QXcbIntegration(QStringList const&, int&, char**) (this=this@entry=0x555555a57710, parameters=..., argc=@0x7fffffffda4c: 1, argv=argv@entry=0x7fffffffdd78)
    at ../../../../include/QtCore/../../src/corelib/tools/qscopedpointer.h:138
#7  0x00007ffff7fbb510 in QXcbIntegrationPlugin::create(QString const&, QStringList const&, int&, char**)
    (this=<optimized out>, system=<optimized out>, parameters=..., argc=@0x7fffffffda4c: 1, argv=0x7fffffffdd78) at ./src/plugins/platforms/xcb/qxcbmain.cpp:56
#8  0x00007fffee26a6db in init_platform (argv=<optimized out>, argc=<optimized out>, platformThemeName=..., platformPluginPath=..., pluginNamesWithArguments=...) at kernel/qguiapplication.cpp:1223
#9  QGuiApplicationPrivate::createPlatformIntegration() (this=0x555555a56520) at kernel/qguiapplication.cpp:1481
#10 0x00007fffee26bc18 in QGuiApplicationPrivate::createEventDispatcher() (this=<optimized out>) at kernel/qguiapplication.cpp:1498
#11 0x00007fffeda70785 in QCoreApplicationPrivate::init() (this=this@entry=0x555555a56520) at kernel/qcoreapplication.cpp:835
#12 0x00007fffee26eb20 in QGuiApplicationPrivate::init() (this=this@entry=0x555555a56520) at kernel/qguiapplication.cpp:1527
#13 0x00007fffee98056d in QApplicationPrivate::init() (this=0x555555a56520) at kernel/qapplication.cpp:513
#14 0x00007fffee980650 in QApplication::QApplication(int&, char**, int) (this=this@entry=0x7fffffffdaa0, argc=<optimized out>, argv=<optimized out>, _internal=_internal@entry=331524)
    at kernel/qapplication.cpp:501
#15 0x0000555555777d64 in SharedTools::QtSingleApplication::QtSingleApplication(QString const&, int&, char**)
    (this=0x7fffffffdaa0, appId=..., argc=<optimized out>, argv=<optimized out>, this=<optimized out>, appId=<optimized out>, argc=<optimized out>, argv=<optimized out>)
    at ./src/3rdparty/qtsingleapplication/qtsingleapplication.cpp:55
#16 0x00005555556934af in OCC::Application::Application(int&, char**) (this=0x7fffffffdaa0, argc=@0x7fffffffda4c: 1, argv=0x7fffffffdd78, this=<optimized out>, argc=<optimized out>, argv=<optimized out>)
    at ./src/gui/application.cpp:201
#17 0x00005555555fd027 in main(int, char**) (argc=<optimized out>, argv=<optimized out>) at ./src/gui/main.cpp:105
```

the crash happens when constructing QApplication.
It could be `QCoreApplication::setAttribute(Qt::AA_UseOpenGLES)` , don't know how this works if qt runs on X11, maybe a Qt bug (that will not be fixed in qt5 since that is essentially eol, at least the free version)

**Comment by akarl10** [comment]
acually, I was wrong, its simply gcc's lto that does break things.. in the current changelog of the package lto is mentioned to be disables, but it seems it isn't...

**Comment by pgdooley** [comment]
Yes I agree - it seems that 3.6.0-1ubuntu1 should have LTO AGAIN disabled, but I suspect it hasn't...I have raised a bug on launchpad so hopefully someone will sort it out.

**Comment by pgdooley** [comment]
Seems to be fixed in   3.6.50-20220912.144217.88464a097-1.0~kinetic1 (available in alpha repository)
